sunpy/net/jsoc/jsoc.py

```diff
@@ -43,10 +43,20 @@ def __init__(self, table=None):
         """
         table : `astropy.table.Table`
         """
-
-        self.table = table
-        self.query_args = None
-        self.requests = None
+        if isinstance(table, list):
```

Comment:
I am not sure why you are checking for `list` here, I think what you want to be doing is checking for `JSOCResponse`. Like this:

```suggestion
        if isinstance(table, JSOCResponse):
            self.table = new_table.table
            self.query_args = new_table.query_args
            self.requests = new_table.requests
        else:
            self.table = table
            self.query_args = None
            self.requests = None
```

(there should only be one `else` block.

Reply:
Hi ! sorry for the delay. I try to check for a JSOCResponse but in fido_factory.py in the lines:
```
if isinstance(record_slice, int):
    resp = [client_resp[record_slice]]
else:
    resp = client_resp[record_slice]
# Reconstruct a response object with the sub-indexed records.
ret = resp_type(resp)
```
create a list, and then in if I use `isinstance(table, JSOCResponse)` this will be ignore.